File: 1721-SwappingNodesinaLinkedList/1721-SwappingNodesinaLinkedList.py
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        current = head
        dummy = ListNode(val=0)
        dummy.next = head
        
        slow = head
        fast = head

        for _ in range(k - 1):
            fast = fast.next

        first_node = fast

        fast = fast.next

        while fast:
            slow = slow.next
            fast = fast.next
        
        first_node.val, slow.val = slow.val, first_node.val

        return dummy.next

File: 1721-SwappingNodesinaLinkedList/1721-SwappingNodesinaLinkedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */
var swapNodes = function(head, k) {
    let slow = head;
    let fast = head;
    let dummy = new ListNode()
    dummy.next = head;

    for(let i = 0; i < k - 1; i++){
        fast = fast.next;
    }

    let firstNode = fast;

    fast = fast.next;

    while(fast){
       fast = fast.next;
       slow = slow.next;
    }
    let temp = firstNode.val;
    firstNode.val = slow.val;
    slow.val = temp;


    return dummy.next
};
